Branch-SDK/src/io/branch/referral/BranchUtil.java

```diff
@@ -0,0 +1,66 @@
+package io.branch.referral;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Iterator;
+
+/**
+ * Class for Branch utility methods
+ */
+class BranchUtil {
+
+    /**
+     * Converts a given link param as {@link JSONObject} to string
+     *
+     * @param params Link param JSONObject.
+     * @return A {@link String} representation of link params.
+     */
+    public static String getJsonParamString(JSONObject params) {
+        return stringifyParams(filterOutBadCharacters(params));
+    }
+
+    /**
+     * Convert the given JSONObject to string and adds source value as
+     *
+     * @param params JSONObject to convert to string
+     * @return A {@link String} value representing the JSONObject
+     */
+    public static String stringifyParams(JSONObject params) {
+        if (params == null) {
```

Comment:
There are methods in branch which uses this method without calling `filterOutBadCharacters`.
For eg. `generateShortLink(null, LINK_TYPE_UNLIMITED_USE, 0, null, null, null, null,BranchUtil.stringifyParams(null), callback, true)`